[core]/es_extended/client/modules/markers.lua

```diff
@@ -0,0 +1,101 @@
+local Markers = {}
+
+---@func fun(coords: vector3, radius: number, data?: any)
+function ESX.CreateMarker(data) 
+    local coords = data.coords
+
+    if type(coords) ~= 'vector3' then 
+        if coords.x and coords.y and coords.z then 
+            data.coords = vec3(coords.x, coords.y, coords.z)
+        elseif coords[1] and coords[2] and coords[3] then 
+            data.coords = vec3(coords[1], coords[2], coords[3])    
+        end
+    end
+
+    local Index = #Markers+1
+    Markers[Index] = data
+    return Markers[Index]
+end
+
+function ESX.GetNearestMarker()
+    local Nearest
+    local Distance = 9999
+    for i = 1, #Markers do 
+        local Marker = Markers[i]
+        local dist = Marker.distance
+        if dist < Distance then 
+            Distance = dist
+            Nearest = Marker
+        end 
+    end 
+    return Nearest, Distance
+end 
+
+function ESX.GetMarkersInRange(min, max)
+    local List = {}    
+    for i = 1, #Markers do 
+        local Marker = Markers[i]
+        local Distance = Marker.distance
+        if Distance >= min and Distance <= max then 
+            List[#List+1] = Marker
+        end
+    end 
+
+    return List
+end
+
+function Markers:Run(index)
+    CreateThread(function()
+        local Marker = self[index]
+        while Marker.running do 
+            Wait(0)
+            Marker:Inside()
+        end 
+    end)
+end 
+
+CreateThread(function()
+    while true do 
+        Wait(1500)
+        local Coords = GetEntityCoords(ESX.PlayerData.Ped)
+        for i = 1, #Markers do
+            local Marker = Markers[i]
+            Marker.distance = #(Coords - Marker.coords) 
+            if Marker.distance < Marker.radius then 
+                if Marker.status and Marker.Inside and not Marker.running then 
+                    Marker.running = true
+                    Markers:Run(i)
```

Comment:
hm, if i have 5 markers i created 5 thread?  I think we should avoid this and run code once thread.

Reply:
and please change all variable is correct lowerCamelCase. Thanks

Reply:
Fixed both brother